python/lsst/jointcal/jointcal.py

```diff
@@ -1235,7 +997,7 @@ def _do_load_refcat_and_fit(self, associations, defaultFilter, center, radius,
             On-sky radius to load from reference catalog.
         name : `str`
             Name of thing being fit: "astrometry" or "photometry".
-        refObjLoader : `lsst.meas.algorithms.LoadReferenceObjectsTask`
+        refObjLoader : `lsst.meas.algorithms.ReferenceObjectLoaderBase`
```

Comment:
Ah, shoot, I thought the base class defined the necessary API, but it actually doesn't. It probably should; I don't know if we plan to keep that gen2/3 compatibility base task or not (one could argue that a different gen3 refcat loader could be built using the same API).

Reply:
The base and derived class should probably be consolidated. I don't think anything would be referring to be base on its own (because as you have noticed, it doesn't actually do anything useful).

Reply:
I wrote that base class! It does something very useful: gave the gen3 loader functionality that it was missing and got rid of a lot of bugs in it! I think the problem in using it to define an API was that the gen2/gen3 loaders didn't share an API in their e.g. `LoadPixelBox` methods. I could see using it to define a gen3 refloader API having some utility.

Reply:
It definitely needed to exist as long as gen2 was around.  My point is that the need for a split between base and derived is probably no longer necessary and the two could be spliced together.